static/app/components/dashboards/widgetQueriesForm.tsx

```diff
@@ -108,8 +110,23 @@ class WidgetQueriesForm extends React.Component<Props> {
                   projectIds={selection.projects}
                   query={widgetQuery.conditions}
                   fields={[]}
-                  onSearch={this.handleFieldChange(queryIndex, 'conditions')}
-                  onBlur={this.handleFieldChange(queryIndex, 'conditions')}
+                  onSearch={field => {
+                    // SearchBar will call handlers for both onSearch and onBlur
+                    // when selecting a value from the autocomplete dropdown. This can
+                    // cause state issues for the search bar in our use case. To prevent
+                    // this, we set a timer in our onSearch handler to block our onBlur
+                    // handler from firing if it is within 200ms, ie from clicking an
+                    // autocomplete value.
+                    this.blurTimeout = setTimeout(() => {
+                      this.blurTimeout = undefined;
+                    }, 200);
+                    return this.handleFieldChange(queryIndex, 'conditions')(field);
+                  }}
+                  onBlur={field => {
+                    if (!this.blurTimeout) {
+                      this.handleFieldChange(queryIndex, 'conditions')(field);
```

Comment:
Hmm, why do we even need to do this on blur? Why aren't we doing this on change?

Reply:
Hey Evan, `handleFieldChange` causes our widget chart to query and rerender. Right now our expected behaviour is to query and rerender when the user initiates a search (ie hits enter or selects an autocomplete) or when the user unfocuses the search bar. I don't think we can use `onChange` here because that would essentially cause the widget chart to requery and rerender any time the user types into the search bar.